srtcore/core.cpp

```diff
@@ -465,14 +465,14 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
         if (m_bOpened)
             throw CUDTException(MJ_NOTSUP, MN_ISBOUND, 0);
 
-        if (*(int *)optval <= 0)
+        if (*(const int *)optval <= 0)
             throw CUDTException(MJ_NOTSUP, MN_INVAL, 0);
 
         {
             // This weird cast through int is required because
             // API requires 'int', and internals require 'size_t';
             // their size is different on 64-bit systems.
-            size_t val = size_t(*(int *)optval);
+            const size_t val = size_t(*(const int *)optval);
```

Comment:
const is ok for the variable itself. For the cast it is completely superfluous - you take this constness out immediately even in the same expression by casting to `size_t`.